src/Ninja/G1CP/Content/Testsuite/checks.d

```diff
@@ -0,0 +1,119 @@
+/*
+ * Check if item exists
+ */
+func void G1CP_Testsuite_CheckItem(var string name) {
+    if (!G1CP_IsItemInst(name)) {
+        var string msg; msg = ConcatStrings("Item '", name);
+        msg = ConcatStrings(msg, "' not found");
+        G1CP_TestsuiteErrorDetail(msg);
+        G1CP_TestsuiteStatusPassed = FALSE;
+    };
+};
+
+/*
+ * Check if item exists and return its symbol index
+ */
+func int G1CP_Testsuite_GetItemId(var string name) {
+    // Check if item instance exists
+    G1CP_Testsuite_CheckItem(name);
+
+    // Return the item's symbol index
+    return MEM_GetSymbolIndex(name);
+};
+
+/*
+ * Check if integer variable exists
+ */
+func void G1CP_Testsuite_CheckIntVar(var string name, var int arrIdx) {
+    if (!G1CP_IsIntVar(name, arrIdx)) {
+        var string msg; msg = ConcatStrings("Integer variable '", name);
+        msg = ConcatStrings(msg, "' not found");
+        G1CP_TestsuiteErrorDetail(msg);
+        G1CP_TestsuiteStatusPassed = FALSE;
+    };
+};
+
+/*
+ * Check if integer constant exists
+ */
+func void G1CP_Testsuite_CheckIntConst(var string name, var int arrIdx) {
+    if (!G1CP_IsIntConst(name, arrIdx)) {
+        var string msg; msg = ConcatStrings("Integer constant '", name);
+        msg = ConcatStrings(msg, "' not found");
+        G1CP_TestsuiteErrorDetail(msg);
+        G1CP_TestsuiteStatusPassed = FALSE;
+    };
+};
+
+/*
+ * Check if info dialog exists
+ */
+func void G1CP_Testsuite_CheckInfo(var string name) {
+    // Check if info instance exists
+    if (!G1CP_IsInfoInst(name)) {
+        var string msg; msg = ConcatStrings("Info instance '", name);
+        msg = ConcatStrings(msg, "' not found");
+        G1CP_TestsuiteErrorDetail(msg);
+        G1CP_TestsuiteStatusPassed = FALSE;
+    };
+
+    // Check if info instance is available in info manager
+    if (!G1CP_GetInfo(name)) {
+        var string msg; msg = ConcatStrings("Info instance '", name);
+        msg = ConcatStrings(msg, "' not available");
+        G1CP_TestsuiteErrorDetail(msg);
+        G1CP_TestsuiteStatusPassed = FALSE;
+    };
+};
+
+/*
+ * Check if info dialog exists and return its symbol index
+ */
+func int G1CP_Testsuite_GetInfoId(var string name) {
+    // Check if info instance exists
+    G1CP_Testsuite_CheckInfo(name);
+
+    // Return the dialog's symbol index
+    return MEM_GetSymbolIndex(name);
+};
+
+/*
+ * Check if function exists
+ */
+func void G1CP_Testsuite_CheckFunc(var string name) {
+    if (!G1CP_IsFunc(name, "int|none")) {
```

Comment:
The second argument here is the expected signature of the function (see comments in [func.d](../../blob/master/src/Ninja/G1CP/Content/Symbols/func.d)). As it is right now, it will only allow functions with an integer return type and no parameters.

I think we can have one general function with `func void G1CP_Testsuite_CheckFunc(string name, string signature)` and then special functions for reoccurring fixes: E.g. `G1CP_Testsuite_CheckDialogFunc` with ("void|none") and `G1CP_Testsuite_CheckDialogConditionFunc` with ("int|none") and maybe some more.